addons/purchase/tests/test_purchase_downpayment.py

```diff
@@ -0,0 +1,686 @@
+            'product_id': cls.service_order.id,
+            'product_qty': 2,
+            'product_uom': cls.service_order.uom_id.id,
+            'price_unit': 100,
+            'order_id': cls.purchase_order.id,
+            'taxes_id': False,
+        })
+        cls.pol_product_deliver = cls.env['purchase.order.line'].create({
+            'name': cls.product_deliver.name,
+            'product_id': cls.product_deliver.id,
+            'product_qty': 2,
+            'product_uom': cls.product_deliver.uom_id.id,
+            'price_unit': 100,
+            'order_id': cls.purchase_order.id,
+            'taxes_id': False,
+        })
+
+        cls.expense_account = cls.company_data['default_account_expense']
+        cls.payable_account = cls.company_data['default_account_payable']
+
+        cls.context = {
+            'active_model': 'purchase.order',
+            'active_ids': [cls.purchase_order.id],
+            'active_id': cls.purchase_order.id,
+            'default_journal_id': cls.company_data['default_journal_purchase'].id,
+        }
+
+    @classmethod
+    def create_tax(cls, amount, values=None):
+        vals = {
+            'name': f'Tax {amount} {uuid.uuid4()}',
+            'amount_type': 'percent',
+            'amount': amount,
+            'type_tax_use': 'purchase',
+            'repartition_line_ids': [
+                Command.create({'document_type': 'invoice', 'repartition_type': 'base'}),
+                Command.create({'document_type': 'invoice', 'repartition_type': 'tax', 'account_id': cls.tax_account.id}),
+                Command.create({'document_type': 'refund', 'repartition_type': 'base'}),
+                Command.create({'document_type': 'refund', 'repartition_type': 'tax', 'account_id': cls.tax_account.id}),
+            ]
+        }
+        if values:
+            vals.update(values)
+        return cls.env['account.tax'].create(vals)
+
+    @classmethod
+    def make_downpayment(cls, **kwargs):
+        payment_params = {
+            'advance_payment_method': 'percentage',
+            'amount': 50,
+            **kwargs,
+        }
+        downpayment = cls.env['purchase.advance.payment.wizard'].with_context(cls.context).create(payment_params)
+        downpayment.create_invoices()
+        cls.purchase_order.button_confirm()
+
+    @classmethod
+    def post_bills(cls):
+        cls.purchase_order.invoice_ids.write({'invoice_date': datetime.datetime.today()})
+        cls.purchase_order.invoice_ids.action_post()
+
+    def _assert_invoice_lines_values(self, lines, expected):
+        return self.assertRecordValues(lines, [dict(zip(expected[0], x)) for x in expected[1:]])
+
+    def _check_order_search(self, orders, domain, expected_result):
+        domain += [('id', 'in', orders.ids)]
+        result = self.env['purchase.order'].search(domain)
+        self.assertEqual(result, expected_result, "Unexpected result on search orders")
+
+    def test_downpayment(self):
```

Comment:
Based on the tests that follows, seems like this should be named:

`test_downpayment_fixed_tax_incl`

Although am I missing something or don't all of these tests have tax included anyways because of how the class setup is designed?